lib/serviceClient.ts

```diff
@@ -169,10 +175,60 @@ export class ServiceClient {
    * @param {WebResource} httpRequest - The HTTP request to populate and then to send.
    * @param {operationSpec} operationSpec - The OperationSpec to use to populate the httpRequest.
    */
-  async sendOperationRequest(httpRequest: WebResource, operationSpec: OperationSpec): Promise<HttpOperationResponse> {
+  async sendOperationRequest(httpRequest: WebResource, operationArguments: OperationArguments, operationSpec: OperationSpec): Promise<HttpOperationResponse> {
     httpRequest.method = operationSpec.httpMethod;
     httpRequest.operationSpec = operationSpec;
 
+    if (operationSpec.headerParameters) {
+      for (const headerParameter of operationSpec.headerParameters) {
+        const parameterName: string = headerParameter.parameterName;
+        const headerName: string = headerParameter.headerName || parameterName;
+        const parameterType: OperationParameterType | undefined = headerParameter.type;
+
+        let headerValue: any = operationArguments.arguments[parameterName];
+        if (headerValue != undefined) {
+          if (parameterType != undefined) {
```

Comment:
does a missing parameterType indicate a Swagger/generation time error?

Reply:
i think when we eventually go the parameter mapper route. If a required parameter is not provided then it will throw an error during serialization.